src/http/services/sync.service.ts

```diff
@@ -93,6 +93,18 @@ export class SyncService {
                     sync.type = DeviceTypes.Garage;
                     sync.traits.push(Trait.OpenClose);
                     break;
+				case 'lock':
+					sync.type = DeviceTypes.Lock;
+					sync.traits.push(Trait.LockUnlock);
+					break;
+				case "fan":
+					sync.type = DeviceTypes.Fan
+					sync.traits.push(Trait.OnOff);
+					if (device.fanSpeedControl){
+						sync.traits.push(Trait.FanSpeed);
+						sync.attributes = { availableFanSpeeds: uniq(device.availableFanSpeeds.join(',')};
```

Comment:
from the documentation, availableFanSpeeds is an array of objects that contains name, values, etc.
https://developers.google.com/assistant/smarthome/traits/fanspeed#device-attributes